node/consensus/src/evaluation.rs

```diff
@@ -56,12 +55,19 @@ error_chain! {
 
 /// Attempt to evaluate a substrate block as a node block, returning error
 /// upon any initial validity checks failing.
-pub fn evaluate_initial(
+pub fn evaluate_initial<Block: BlockT, Hash>(
 	proposal: &Block,
 	now: Timestamp,
 	parent_hash: &Hash,
-	parent_number: BlockNumber,
-) -> Result<CheckedBlock> {
+	parent_number: <<Block as BlockT>::Header as HeaderT>::Number,
+) -> Result<CheckedBlock>
+where
+	<<Block as BlockT>::Header as HeaderT>::Number: Add<u64, Output=<<Block as BlockT>::Header as HeaderT>::Number>,
+	<<Block as BlockT>::Header as HeaderT>::Number: PartialEq<u64>,
+	<<Block as BlockT>::Header as HeaderT>::Number: Into<u64>,
```

Comment:
These 3 bounds for `u64` above are not required probably. You can get away with using `As<u64>` which `Number` already supports.

Reply:
Ahh nice, ty. Did not know the `As` trait.